### PR TITLE
[iOS] - Leo QA Iteration 3

### PR DESCRIPTION
**Pull request by Brandon-T**
- Fixing AI-Chat bugs in the SDK where it does not save the user's model selection no matter what
- Potentially fixing bugs in the SDK where the user is somehow premium but still rate-limited

Resolves https://github.com/brave/brave-browser/issues/36824
Resolves  https://github.com/brave/brave-browser/issues/36816
Resolves  https://github.com/brave/brave-browser/issues/36856
Resolves  https://github.com/brave/brave-browser/issues/36857

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by soner-yuksel — APPROVED** [review]

File: ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift [hunk 1]
```diff
@@ -163,6 +169,10 @@ public class AIChatViewModel: NSObject, ObservableObject {
   // This function should not exist
   // We should not be refreshing credentials in this model at all!
   // This should be done in Brave-Skus-Manager once VPN moves to Skus v2
+  // Note: There is another bug where this function is required due to SkusSDK being broken
+  // See: https://bravesoftware.slack.com/archives/C021S0FN3V1/p1709148056615289?thread_ts=1709068050.951029&cid=C021S0FN3V1
```
Comment (iccub) [inline ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift:173]: We should not link to slack in our public repo

File: ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift [hunk 2]
```diff
@@ -98,13 +98,19 @@ public class AIChatViewModel: NSObject, ObservableObject {
...
-    currentModel = api.currentModel
     models = api.models
     conversationHistory = api.conversationHistory
     suggestedQuestions = api.suggestedQuestions
     apiError = api.currentAPIError
     requestInProgress = api.isRequestInProgress
 
+    if let defaultModelKey = Preferences.AIChat.defaultAIModelKey.value {
+      currentModel = api.models.first(where: { $0.key == defaultModelKey }) ?? api.currentModel
+      changeModel(modelKey: defaultModelKey)
```
Comment (petemill) [inline ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift:109]: `ConversationDriver` has its own mechanism for setting the default model from pref or varying via Premium status, which _could_ be async and race with this. At the moment it's not likely to be a bad race condition but that could change in the future. Could you at least set a TODO here to move to the shared pref for default model? We should sync it anyway...

Comment (Brandon-T) [reply ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift]: Fixed. I exported the preference here: https://github.com/brave/brave-core/pull/22627/commits/64dd821cc41128fc1ba0177800a31aa1df2fd737

**Comment by kjozwiak** [comment]
Verification **PASSED** on `iPhone 11` running `iOS 17.4` using the following build(s):

```
1.66.10 Chromium: 123.0.6312.46 (Official Build) unknown (64-bit)
--- | ---
Revision | 741e5a4ef519...
OS | iOS
```

### `Test Case #1` - https://github.com/brave/brave-browser/issues/36824

The following issue can't really be verified on `Nightly` and requires an RC. As per the conversation via https://bravesoftware.slack.com/archives/C04KPE0BUDU/p1710518963710699, we'll need to create new accounts to run through the above and ensure that `Premium` users aren't getting rate limited. @Uni-verse @hffvld and myself will discuss and run through three different scenarios that involve the `Rate Limit` modal.

### `Test Case #2` - https://github.com/brave/brave-browser/issues/36816

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/36816#issue-2187708393, ensured the following:

* ensured that switching between `Mixtral`, `Claude` and `Lama` via `Advanced Settings` retained the users selection
* ensured that switching the default LLM via `Leo` panel didn't change the global setting
  * Example: `Claude` set as default and then selecting `Lama` as a model for the tab instance (shouldn't change global setting)

https://github.com/brave/brave-core/assets/2602313/fe6383f0-11d6-45f8-bcaa-7dc1fc1dba94

However, running through the above, it looks like the global setting is not retained/remembered when restarting `Brave` and `Mixtral` is always selected as the default. The above fixed the issue re: same session but didn't fix the issue when restarting Brave. Created https://github.com/brave/brave-browser/issues/36886 as a follow up that should be fixed for `1.63.x`.

### `Test Case #3` - https://github.com/brave/brave-browser/issues/36856

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/36856#issue-2189351577, ensured that the string/text isn't being cut off on certain locales such as `German` & `Russian`. As per the conversation via https://bravesoftware.slack.com/archives/C05NDQP03AQ/p1710531689040729, we're going to live with the string taking up two lines rather than a single line for the initial release but will address the issues via https://github.com/brave/brave-browser/issues/36870 & https://github.com/brave/brave-browser/issues/36871.

**`German`**

**`Example`** | **`Example`**
----------------|----------------
IMG_0137 | IMG_0138

**`Russian`**

**`Example`** | **`Example`**
----------------|----------------
IMG_0139 | IMG_0140

### `Test Case #4 ` - https://github.com/brave/brave-browser/issues/36857

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/36857#issue-2189413192, ensured that `Add Feedback` was visible for `Dislike` via both `Light` & `Dark` themes as per the following:

**`Disliked`**

**`Example`** | **`Example`**
----------------|----------------
IMG_0141 | IMG_0143

**`Liked`**

**`Example`** | **`Example`**
----------------|----------------
IMG_0142 | IMG_0144

**Pull request merged**